lib/core/src/sdk.rs

```diff
@@ -56,7 +56,50 @@ pub struct LiquidSdk {
 
 impl LiquidSdk {
     pub async fn connect(req: ConnectRequest) -> Result<Arc<LiquidSdk>> {
-        let config = req.config;
+        // If we can fetch a proxy swapper URL, extract referral ID and dynamically set it in the config
+        let (boltz_api_base_url, referral_id) = match &req.config.network {
+            LiquidNetwork::Testnet => (None, None),
+            LiquidNetwork::Mainnet => {
+                match BreezServer::new("https://bs1.breez.technology:443".into(), None) {
+                    Ok(breez_server) => match breez_server.fetch_boltz_swapper_urls().await {
+                        Ok(boltz_swapper_urls) => match boltz_swapper_urls.first().cloned() {
+                            Some(first_url) => Url::parse(first_url.as_str())
+                                .map(|url| match url.query() {
+                                    None => (None, None),
+                                    Some(query) => {
+                                        let boltz_api_base_url = url
+                                            .domain()
+                                            .map(|domain| format!("https://{domain}/v2"));
+                                        let parts: Vec<String> =
+                                            query.split('=').map(Into::into).collect();
+                                        let referral_id = parts.get(1).cloned();
```

Comment:
I think it is better to save the query as is and append it to the create requests, not need to parse it for the specific param.